libraries/AP_BattMonitor/AP_BattMonitor_UAVCAN.cpp

```diff
@@ -209,7 +209,11 @@ void AP_BattMonitor_UAVCAN::handle_mppt_stream(const MpptStreamCb &cb)
         mppt_set_bootup_powered_state();
     }
 
+#if AP_BATTMONITOR_UAVCAN_MPPT_FAULT_ANNOUNCE
     mppt_check_and_report_faults(cb.msg->fault_flags);
+#else 
+    _mppt.fault_flags = cb.msg->fault_flags;
```

Comment:
I'm not really comfortable with setting this in two places.  Could be avoided by keeping that method and cutting out the bits that aren't needed.

If nothing else a comment in here that it's normally set by the method would be nice.

Not a blocker, and you're the maintainer, so if you want this in as-is I'll merge it.

Reply:
I agree.  maintaining the state in two places is a common mistake we make that adds complexity.  Its much better to set the state in a single place.  This is why in Copter, for example, you'll see methods like "update_land_detector" while in messy features you'll see a state variable updated in multiple methods across various .cpp files.

Reply:
good points. I'll localize how it does this so it's only setting ```_mppt.fault_flags``` once